ast/parser_test.go

```diff
@@ -2400,6 +2400,102 @@ func TestNoMatchError(t *testing.T) {
 	}
 }
 
+func TestBraceBracketParenMatchingErrors(t *testing.T) {
+	// Checks to prevent regression on issue #4672.
+	// Error location is important here, which is why we check
+	// the error strings directly.
+
```

Comment:
There are tests of various styles in this file. I think the preferred style today is something like https://github.com/open-policy-agent/opa/blob/193ce0465e7b7e7866f5f18b31ff3ea79e5c34e2/ast/parser_test.go#L3649. define a slice of structs, loop over it with t.Run. that way, you can optionally execute just one test out of many.

And there are a bunch of helpers that we could use for asserting stuff on errors. assertParseError etc at the bottom of this file.

Reply:
Ah! I had wondered about that. I'll refactor to use the loop style right quick, and then ping back for the merge.

Reply:
@srenatus Refactored to use the slice-of-structs style. For now, I'm just asserting on the raw error text, since that catches the majority of what I wanted to catch with these new tests.